### ISSUE TITLE
Autowidth does not work

### ISSUE THREAD

**Issue by marcushellberg** [vaadin/vaadin-grid-flow#685]
Vaadin 14 beta3. Chrome 75/mac.

Grid before applying auto width:
<img width="788" alt="Screen Shot 2019-06-18 at 6 57 03 PM" src="https://user-images.githubusercontent.com/883698/59731492-35fb6f00-91fb-11e9-907f-29d790ab6232.png">

after calling `getColumns().forEach(column -> column.setAutoWidth(true));`
<img width="784" alt="Screen Shot 2019-06-18 at 6 58 10 PM" src="https://user-images.githubusercontent.com/883698/59731518-4a3f6c00-91fb-11e9-8163-31ed17b9d0e1.png">

Expected result: 
I expected that after enabling auto-width, the email column would have displayed full emails as there is enough space

**Comment by Haprog** [comment]
Please read the documentation for "Column auto width". It clearly says (in WC API docs, WC demo and Flow API docs) that column widths are calculated automatically only once when the grid items are rendered for the first time. If the grid has been rendered already when setting auto width you need to explicitly call `Grid#recalculateColumnWidths()` to trigger recalculation. It works as intended.

**Comment by Haprog** [comment]
@marcushellberg I opened a new issue here to make this DX slightly better (and to make it work like you expected): https://github.com/vaadin/vaadin-grid/issues/1629

**Comment by marcushellberg** [comment]
The only documentation I could find was on the release page, and that didn't mention recalculating https://vaadin.com/releases/vaadin-14#grid-autowidth-column

**Comment by Legioth** [comment]
It's documented through the latest version of the demo, but vaadin.com is still using the Vaadin 13 version.

https://github.com/vaadin/vaadin-grid-flow/blob/f41d74e8f7ec6c4d809d0887bc69bba3a4b58b8f/vaadin-grid-flow-demo/src/main/java/com/vaadin/flow/component/grid/demo/GridDemo.java#L1210-L1219

**Comment by Haprog** [comment]
https://github.com/vaadin/vaadin-grid-flow/blob/master/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java#L460

See also JavaDocs of setAutoWidth.
